core/mo-services-impl/ccsds-com-impl/src/main/java/esa/mo/com/impl/provider/ArchiveSyncProviderServiceImpl.java

```diff
@@ -79,463 +79,645 @@
-        objInstIds.addAll(ids);
-        try {
-          Thread.sleep(1000);
-          LongList deleted = archive.getArchiveStub().delete(type, domain, objInstIds);
-          if(quota != null){
-            quota.clean(clearedIds);
-          }
-        } catch (MALInteractionException ex) {
-          Logger.getLogger(ArchiveSyncProviderServiceImpl.class.getName()).log(Level.SEVERE, null,
-              ex);
-        } catch (MALException ex) {
-          Logger.getLogger(ArchiveSyncProviderServiceImpl.class.getName()).log(Level.SEVERE, null,
-              ex);
-        } catch (InterruptedException ex) {
-          Logger.getLogger(ArchiveSyncProviderServiceImpl.class.getName()).log(Level.SEVERE, null,
-              ex);
+        public Runnable getFlushingRunnable()
+        {
+            return new Runnable()
+            {
+                @Override
+                public void run()
+                {
+                    try
+                    {
+                        boolean exit = false;
+                        int index = 0;
+                        byte[] aChunk = new byte[chunkSize];
+                        int pos = 0;
+
+                        while (!exit)
+                        {
+                            boolean done = processingIsDone;
+                            byte[] data = dataToFlush.poll(pollTiemout, TimeUnit.MILLISECONDS);
+
+                            if (data != null)
+                            {
+                                for (int i = 0; i < data.length; i++)
+                                {
+                                    aChunk[pos] = data[i];
+                                    pos++;
+
+                                    if (pos == chunkSize)
+                                    {
+                                        sendUpdateToConsumer(index, aChunk);
+                                        index++;
+                                        pos = 0;
+                                    }
+                                }
+                            }
+                            else
+                            {
+                                if (done)
+                                {
+                                    exit = true;
+
+                                    // Flush the last byte array!
+                                    byte[] lastChunk = new byte[pos]; // We need to trim to fit!
+                                    System.arraycopy(aChunk, 0, lastChunk, 0, pos);
+
+                                    sendUpdateToConsumer(index, lastChunk);
+                                    index++;
+                                }
+                            }
+                        }
+
+                        try
+                        {
+                            interaction.sendResponse(new UInteger(index));
+                            numberOfChunks = index;
+                        }
+                        catch (MALInteractionException ex)
```

Comment:
Please don't blow up the throw-catches like that, you are not being paid by the number of lines coded, so no need to increase the number of lines in the files. It makes it harder to read because one needs to scroll through it!
Use: 
try{
abc
}catch (MALException ex){
xyz
}

same applies to the ifs.

Reply:
Note that the commited code actually adheres to ESOC Java Style, so it's good. I plan to apply it globally onto NMF eventually.